pkg/controller/operators/catalog/operator.go

```diff
@@ -202,6 +204,12 @@ func NewOperator(ctx context.Context, kubeconfigPath string, clock utilclock.Clo
 		// Wire Subscriptions
 		subInformer := crInformerFactory.Operators().V1alpha1().Subscriptions()
 		op.lister.OperatorsV1alpha1().RegisterSubscriptionLister(namespace, subInformer.Lister())
+		if err := subInformer.Informer().AddIndexers(cache.Indexers{index.PresentCatalogIndexFuncKey: index.PresentCatalogIndexFunc}); err != nil {
```

Comment:
An indexer takes an object and returns a set of keys that can be used to find that same object again in the future.

For this case, don't we want to add an indexer to namespaces that let's us find the namespaces to requeue? 

I.e. we want: `in: CatalogSource`, `out: Namespaces`, and then we requeue each of those namespaces.

Currently, this indexer is configured for:

`in: CatalogSource, out: Subscriptions` and we have to iterate over the subscriptions to get the list of namespaces.

it would be nice if our processing was just: 

```go
case connectivity.Ready:
  namespaces  := op.catalogSubscriberIndexer[state.key.Namespace].ByIndex(PresentCatalogIndexFuncKey, fmt.Sprintf("%s/%s", state.key.Name, state.key.Namespace))

  for _, ns := range namespaces {
     o.nsResolveQueue.Add(ns) 
  }
case ...:
```

Reply:
Hey Evan. I did, in fact, do `in: CatalogSource, out: Namespaces` in previous iteration of this PR but changed to current `in: CatalogSource, out: Subscriptions` implementation. The reason for it is I think it is more valuable to find out the list of `Subscriptions` that subscribes to a specific `CatalogSource` than the list of `namespaces`. Also, with this current implementation, the `namespaces` are in values of the returned map. In the way, we still have direct access to the list of namespaces without any extra works. There is a complication of duplicated namespaces but the `workerqueue` is using a map behind the scene and it automatically rejects duplicated item by default.

Reply:
Either way, if you feel there is no value in get `Subscriptions` list then I can simplify it to just get a `namespaces` list.